lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -640,6 +640,18 @@ protected function prepare_links( $comment ) {
 			);
 		}
 
+		// Only grab one comment to verify the comment has children.
+		$comment_children = $comment->get_children( array( 'number' => 1 ) );
```

Comment:
Since we only care about the count, we should pass `count => true`, which queries for `COUNT(*)` instead of the fields. This is less expensive when we actually need to hit the database, and should be equivalent performance for the cached case.